client2/poisson_process_test.go

```diff
@@ -0,0 +1,42 @@
+// SPDX-FileCopyrightText: © 2023 David Stainton
+// SPDX-License-Identifier: AGPL-3.0-only
+
```

Comment:
- What are we testing in this file?
  - What do the operations do for test?
  - What is the expected outcome for each test?
  - Why is that outcome expected
documenting this lets us modify this code in the future without spending ages reverse-engineering the logic.

Reply:
ive added commentary to each test